procrastinate/blueprints.py

```diff
@@ -271,11 +324,26 @@ def _wrap(func: Callable[..., tasks.Task]):
             return functools.update_wrapper(task, func, updated=())
 
         if _func is None:  # Called as @app.task(...)
-            return _wrap
+            return cast(
+                Union[
+                    Callable[[Callable[P, Any]], Task[P, P]],
+                    Callable[
+                        [Callable[Concatenate[JobContext, P], Any]],
+                        Task[Concatenate[JobContext, P], P],
+                    ],
+                ],
+                _wrap,
+            )
```

Comment:
I think we don't need this cast anymore

Reply:
I am afraid it won't accept it otherwise. 
I am making it explicit that this is the intended signature of the returned decorator.
Alternatively, I can make it `cast(Any, _wrap)`

If you know a way that doesn't involve a `cast` here, I am all ears!